docs/.vuepress/plugins/markdown-it-conditional-container/index.js

```diff
@@ -0,0 +1,142 @@
+      }
+
+      token.children.splice(childrenIndex, howMany);
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(`${chalk.red('\nUnexpected error thrown while removing conditional container.' +
+        ` Please check how "${env.frontmatter.permalink}" site, parsed from ` +
+        `"${getNormalizedPath(env.relativePath)}" file looks like.`
+      )}`);
+    }
+  };
+  
+  const cleanTokens = ({ tokens, token, tokenIndex, preciseRegexp, lessPreciseRegexp, env }) => {
+    if (token === void 0) {
+      // eslint-disable-next-line no-console
+      console.error(`${chalk.red('\nUnexpected error thrown while removing conditional container.' +
+        ` It is possible that \`::: only-for\' value has been fount, but ending \':::\' value has not.`
+      )}`);
+    }
+
+    if (preciseRegexp.test(token.content)) {
+      tokens.splice(tokenIndex, 1);
+
+    } else {
+      removeValueAndNewLine({ token, regexp: lessPreciseRegexp, env });
+    }
+  };
+
+  const findAndRemove = (state) => {
+    const relativePath = state.env?.relativePath; // Sometimes the `env` key is an empty object.
+
+    if (relativePath === void 0) {
+      return;
+    }
+
+    const env = state.env;
+    const frameworkId = getEnvDocsFramework() || parseFramework(relativePath) || getDefaultFramework();
+
+    for (let index = state.tokens.length - 1; index >= 0; index -= 1) {
+      const token = state.tokens[index];
+      // We don't create custom container intentionally. It can create paragraphs or break listed elements.
+      const isNotNativeContainer = token.markup !== markupForCustomContainer;
+      
+      if (isNotNativeContainer) {
+        if (openAndCloseTagOneliner.test(token.content)) {
+          const onlyForFrameworks = openAndCloseTagOneliner.exec(token.content)[capturedGroupIndex].split(' ');
+
+          if (onlyForFrameworks.includes(frameworkId) === false) {
+            state.tokens.splice(index, 1); // We remove full token containing oneliner.
+
+          } else {
+            removeValueAndNewLine({ token, regexp: openTokenContent, env });
+            removeValueAndNewLine({ token, regexp: closeTokenContent, env });
+          }
+
+        } else if (closeTokenContent.test(token.content)) {
+
+          endIndex = index;
+        } else if (openTokenContent.test(token.content)) {
+          const onlyForFrameworks = openTokenContent.exec(token.content)[capturedGroupIndex].split(' ');
+
+          if (onlyForFrameworks.includes(frameworkId) === false) {
+            const startIndexForRemoval = index + 1; // We remove elements, starting from the token right after the found one.
+            const elementsForRemoval = endIndex - startIndexForRemoval + 1;
+
+            state.tokens.splice(startIndexForRemoval, elementsForRemoval);
+
+            // Token's content may starts with `::: only-for` (should be removed). Alternatively, it can end with the markup.
+            // Then, just string's tail should be removed.
```

Comment:
I am not sure what the above comment means. Can you explain better in the code comment?